src/charm.py

```diff
@@ -443,13 +446,63 @@ def remote_write_endpoints(self):
         """Return remote-write endpoints."""
         return self._remote_write.endpoints
 
+    def _update_source_exchange(self) -> None:
+        """Update the grafana-datasource relations."""
+
+        # This degree of multiplexing requires some in-depth explanation.
+        #   each grafana we're sending our data to gives us back a mapping from unit names to datasource UIDs.
```

Comment:
I think this comment is outdated now

Reply:
why? I think it's still applicable

Reply:
well now we should have the grafana UID as key

Reply:
mmm no that bit is referring to the grafana-source relation. I'll clarify